pyfstat/utils/atoms.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+import lalpulsar
+
+logger = logging.getLogger(__name__)
+
+
+def extract_singleIFOmultiFatoms_from_multiAtoms(
+    multiAtoms: lalpulsar.MultiFstatAtomVector, X: int
+) -> lalpulsar.MultiFstatAtomVector:
+    """Extract a length-1 MultiFstatAtomVector from a larger MultiFstatAtomVector.
+
+    The result is needed as input to ``lalpulsar.ComputeTransientFstatMap`` in some places.
+
+    The new object is freshly allocated,
+    and we do a deep copy of the actual per-timestamp atoms.
+
+    Parameters
+    -------
+    multiAtoms:
+        Fully allocated multi-detector struct of `length > X`.
+    X:
+        The detector index for which to extract atoms.
+    Returns
+    -------
+    singleIFOmultiFatoms:
+        Length-1 MultiFstatAtomVector with only the data for detector `X`.
+    """
+    if X >= multiAtoms.length:
```

Comment:
@dbkeitel 

This was 
```suggestion
    if X > multiAtoms.length:
```
which I'd say was wrong. 

Agree?

Reply:
absolutely